### PR TITLE
[GROUND-1] Clean up Postgres & Cassandra Schemas

### PR DESCRIPTION
**Pull request by vsreekanti**

### CODE REVIEW

**Comment by vsreekanti** [comment]
@tarpdalton Can you look at this? I tried to be as uniform as possible.

**Comment by tarpdalton** [comment]
I'll look at it and ask for hemal's input.

File: ground-core/scripts/cassandra/cassandra.sql [hunk 1]
```diff
@@ -1,134 +1,136 @@
 /* VERSIONS */
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: File ext should be `.CQL` for cassandra

File: ground-core/scripts/cassandra/cassandra.sql [hunk 2]
```diff
@@ -1,134 +1,136 @@
...
 );
 
 /* MODELS */
 
-create table Structures (
+create table structure (
     item_id varchar,
     name varchar,
-    primary key(item_id, name)
+    PRIMARY KEY (item_id, name)
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: Could a structure have more than one name? Feels weird.
(Ground noob here, so this is more a question than a comment)

File: ground-core/scripts/cassandra/cassandra.sql [hunk 3]
```diff
@@ -1,134 +1,136 @@
...
 );
 
-create table StructureVersions (
+create table structure_version (
     id varchar primary key,
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: `PRIMARY KEY` keyword is uppercase in newly edited instances. Change for consistency?

File: ground-core/scripts/cassandra/cassandra.sql [hunk 4]
```diff
@@ -1,134 +1,136 @@
...
-    structure_id varchar,
+create table rich_version (
+    id varchar PRIMARY KEY,
+    structure_version_id varchar,
     reference varchar
 );
 
-create table RichVersionExternalParameters (
-    richversion_id varchar,
+create table rich_version_external_parameter (
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: This feels highly "relational" Could it be folded into the `RichVersions` table? Could maybe use a `Map` structured field in `rich_versions`?

File: ground-core/scripts/cassandra/cassandra.sql [hunk 5]
```diff
@@ -1,134 +1,136 @@
...
+    rich_version_id varchar,
     key varchar,
     value varchar,
-    primary key (richversion_id, key)
+    PRIMARY KEY (rich_version_id, key)
 );
 
-create table Tags (
-    richversion_id varchar,
+create table tag (
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: same comment as for `rich_version_external_parameter`

File: ground-core/scripts/cassandra/cassandra.sql [hunk 6]
```diff
@@ -1,134 +1,136 @@
...
-    primary key(item_id, name)
+    PRIMARY KEY (item_id, name)
 );
 
-create table Graphs (
+create table graph (
     item_id varchar,
     name varchar,
-    primary key(item_id, name)
+    PRIMARY KEY (item_id, name)
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: Could we have more than a `name` ? Weird to have it as part of the `PRIMARY KEY`

File: ground-core/scripts/cassandra/cassandra.sql [hunk 7]
```diff
@@ -1,134 +1,136 @@
...
     edge_id varchar,
-    endpoint_one varchar,
-    endpoint_two varchar,
+    from_node_version_id varchar,
+    to_node_version_id varchar
 );
 
-create table GraphVersions (
+create table graph_version (
     id varchar primary key,
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: `PRIMARY KEY` in uppercase, for consistency?

File: ground-core/scripts/cassandra/cassandra.sql [hunk 8]
```diff
@@ -1,134 +1,136 @@
...
-    primary key(node_id, name)
+    PRIMARY KEY (node_id, name)
 );
 
-create table LineageEdges (
+create table lineage_edge (
     item_id varchar,
     name varchar,
-    primary key(item_id, name)
+    PRIMARY KEY (item_id, name)
```
Comment (maasg) [inline ground-core/scripts/cassandra/cassandra.sql]: Same comment as above about the `name` attribute. Does not seem to belong to the primary key (In Cassandra, it becomes the cluster key, effectively creating an internal structure as: `(item_id, name) -> timestamp` that does not seem to be required)

**Review by maasg — COMMENTED** [review]
Several comments on the Cassandra data model for your consideration. 
As a general comment, it feels like a 1-1 mapping from a relational model, which might not fit Cassandra well and result in lower performance due to the need to many application-level joins, given that Cassandra does not support joins out-of-the-box.  Depending on the Cassandra version used, we could consider using `collections` for some of the fields, in order to reduce the complexity = fewer queries)

(I'm new to Ground, but have considerable Cassandra experience)

File: ground-core/scripts/cassandra/cassandra.sql (hunk 2 above)
Comment (vsreekanti) [reply ground-core/scripts/cassandra/cassandra.sql]: What do you mean by more than one name? `id` is an internal key, and `name` is an external facing name that has to be unique.

**Comment by vsreekanti** [comment]
@maasg Thanks for the feedback, Gerard! Like you noted, it is very much a direct mapping from a relational schema. It would be great if you could point out where all we could make those improvements (or point us to some resources on efficient mappings of relational schema to Cassandra). 

The reason name is included in the primary key of the table was because there's a uniqueness constraint on it, which, as far as I know, Cassandra does not allow. Is there a more canonical way to enforce that?

**Comment by tarpdalton** [comment]
The postgres changes look good

**Pull request merged**